lib/Injections/NetworkAgent.js

```diff
@@ -0,0 +1,306 @@
+// This function will be injected into the target process.
+module.exports = function(require, debug, options) {
+  var http = require('http'),
+      https = require('https'),
+    deferToConnect = http.ClientRequest.prototype._deferToConnect;
+
+  function timestamp() {
+    return Date.now() / 1000000;
+  }
+
+  function Timing() {
+    this._hrtime = process.hrtime();
+
+    this.json = {
+      requestTime: Date.now() / 1000,
+      proxyStart: -1,
+      proxyEnd: -1,
+      dnsStart: -1,
+      dnsEnd: -1,
+      connectStart: -1,
+      connectEnd: -1,
+      sslStart: -1,
+      sslEnd: -1,
+      serviceWorkerFetchStart: -1,
+      serviceWorkerFetchReady: -1,
+      serviceWorkerFetchEnd: -1,
+      sendStart: -1,
+      sendEnd: -1,
+      receiveHeadersEnd: -1
+    };
+  }
+
+  [
+    'proxyStart', 'proxyEnd',
+    'dnsStart', 'dnsEnd',
+    'connectStart', 'connectEnd',
+    'sslStart', 'sslEnd',
+    'serviceWorkerFetchStart', 'serviceWorkerFetchReady', 'serviceWorkerFetchEnd',
+    'sendStart', 'sendEnd',
+    'receiveHeadersEnd'
+  ].forEach(function(method) {
+    Timing.prototype[method] = function() {
+      var diff = process.hrtime(this._hrtime);
+      this.json[method] = diff[0] * 1e3 + diff[1] / 1e6;
+      return this;
+    };
+  });
+
+  function getStackTrace() {
+    var backup = Error.prepareStackTrace;
+
+    Error.prepareStackTrace = function(_, stack) { return stack; };
+    // We want to trim stack trace to display user caller function.
+    // We wait what request will be sent by one of functions in `callers` list.
+    // If function exists in stack trace, his length will be great than 0
+    // If there is no known caller func we sent full stack trace
+    var callers = [http.request, http.get, https.request, https.get],
+      error,
+      stack;
+
+    while (!stack && callers.length) {
+      error = new Error();
+      Error.captureStackTrace(error, callers.pop());
+      if (error.stack.length) stack = error.stack;
+    }
+
+    if (!stack) stack = new Error().stack;
+
+    Error.prepareStackTrace = backup;
+
+    return stack.reduce(function(stack, frame) {
+      var fileName = frame.getFileName();
+
+      // v8NameToInspectorUrl
```

Comment:
Makes me wonder: is it possible to convert the filename to the url in NI process, so that we can reuse `convert.v8NameToInspectorUrl` and don't have to maintain the code in two places?